s3-sink-connector/src/main/java/io/aiven/kafka/connect/s3/config/S3SinkConfig.java

```diff
@@ -626,30 +197,17 @@ public Template getPrefixTemplate() {
     }
 
     public ZoneId getTimezone() {
-        return ZoneId.of(getString(TIMESTAMP_TIMEZONE));
+        return ZoneId.of(getString(S3ConfigFragment.TIMESTAMP_TIMEZONE));
     }
 
     public TimestampSource getTimestampSource() {
-        return TimestampSource.of(getTimezone(), TimestampSource.Type.of(getString(TIMESTAMP_SOURCE)));
-    }
-
-    public AwsStsRole getStsRole() {
-        return new AwsStsRole(getString(AWS_STS_ROLE_ARN), getString(AWS_STS_ROLE_EXTERNAL_ID),
-                getString(AWS_STS_ROLE_SESSION_NAME), getInt(AWS_STS_ROLE_SESSION_DURATION));
-    }
-
-    public boolean hasAwsStsRole() {
-        return getStsRole().isValid();
-    }
-
-    public boolean hasStsEndpointConfig() {
-        return getStsEndpointConfig().isValid();
-    }
-
-    public AwsStsEndpointConfig getStsEndpointConfig() {
-        return new AwsStsEndpointConfig(getString(AWS_STS_CONFIG_ENDPOINT), getString(AWS_S3_REGION_CONFIG));
+        return TimestampSource.of(getTimezone(), TimestampSource.Type.of(getString(S3ConfigFragment.TIMESTAMP_SOURCE)));
     }
 
+    /**
+     * Deprecated please use S3ConfigFragment.AwsRegionValidator
```

Comment:
Because this is a protected class, can we delete it instead of deprecating ?

Reply:
Technically not as it could have been extended by a contributor the best practice is to deprecate. if it was private we could have deleted it though